### ISSUE TITLE
Number of pages does not change when entering search term

### ISSUE THREAD

**Issue by galbwe** [codefordenver/partner-finder#154]
The number of pages displayed in the top right does not update when entering a search term.

To reproduce:
- with no search term entered, move to the second page of results

<img width="396" alt="search-bug-1" src="https://user-images.githubusercontent.com/20004072/136321214-886c3b22-9c4e-488d-b28b-6d3b9fc04f92.png">

- enter the search term "Aurora"

<img width="390" alt="search-bug-2" src="https://user-images.githubusercontent.com/20004072/136321248-cefeef3c-ed37-47f8-ab7a-7d71dff767c4.png">

There are only 9 leads in the database matching this search, so the number of pages should change to 1.

I think it also makes sense to set the active page to 1 when a new search term is entered. This prevents displaying something like "page 2/1" in the top right.

In summary, the new display should be `1/1` after entering the search term "Aurora"

**Comment by galbwe** [comment]
@taimooralqureshi sure thing. Thanks!
